src/actions/util.js

```diff
@@ -7,3 +8,10 @@ export const findError = (errors, type, bucket, id) => {
 
   return get(find([errors[type]], match), 'payload.response')
 }
+
+export const setCurrentCommunityIdLocalAndRemote = (dispatch, communityId, userId) => {
+  if (!communityId) return
+  const settings = {currentCommunityId: communityId}
+  userId && dispatch(updateUserSettings(userId, {settings}))
```

Comment:
btw, i think we have a kind-of-serious bug on the server side where updating one value in settings causes all the other existing values to be dropped. can you fix that and write a test for it?